BWEnv/src/controller.cc

```diff
@@ -773,68 +794,90 @@ void Controller::onFrame() {
   // case.
   bool receive_commands = !last_receive_ok;
 
-  // Send frame out to Lua side
   if (send_frame) {
-    // only done once before sending
-
+    
+    fbs::Vec2 vec2ScreenPosition;
+    fbs::Vec2 vec2VisibilitySize;
+    fbs::Vec2 vec2ImgSize;
+    std::string imgMode;
+    
+    auto sendImageData = false;
     if (with_image_) {
-      this->tcframe_.img_mode = config_->img_mode;
-
-      auto pos = BWAPI::Broodwar->getScreenPosition();
-      this->tcframe_.screen_position->mutate_x(pos.x);
-      this->tcframe_.screen_position->mutate_y(pos.y);
-
-      // get visibility
-      this->tcframe_.visibility_size->mutate_x(20);
-      this->tcframe_.visibility_size->mutate_y(13);
-      this->tcframe_.visibility.resize(20 * 13);
-
-      auto init_x = pos.x / 32;
-      auto init_y = pos.y / 32;
-      auto it = this->tcframe_.visibility.begin();
-      for (auto y = 0; y < 13; y++) {
-        for (auto x = 0; x < 20; x++) {
+      with_image_ = false;
+      std::unique_ptr<std::string> imageData(recorder_->getScreenData(
+        config_->img_mode,
+        config_->window_mode,
+        config_->window_mode_custom));
+      sendImageData = imageData->size() > 0;
+      if (sendImageData) {
+        vec2ImgSize.mutate_x(recorder_->width);
+        vec2ImgSize.mutate_y(recorder_->height);
+        this->image_data_.resize(imageData->size());
+        std::copy(imageData->begin(), imageData->end(), this->image_data_.begin());        
+      }
+      
+      imgMode = config_->img_mode;
+      
+      auto screenPosition = BWAPI::Broodwar->getScreenPosition();
+      vec2ScreenPosition.mutate_x(screenPosition.x);
+      vec2ScreenPosition.mutate_y(screenPosition.y);
+      
+      const int visibilityTileWidth = 20;
+      const int visibilityTileHeight = 13;
+      const int tileSize = 32;    
```

Comment:
Nit: I think we discussed on Messenger what these value represent? It would be good to have better names for them and declare them like `constexpr int kTileSize = 32`, for example.

Reply:
We discussed what they might mean, but nobody including their original author recalled exactly. I fear any further documentation would be just as likely to confuse as to enlighten.

Reply:
Alright, let's keep it as is then.